### PR TITLE
Added utils for polling for existance post creation, before final read into state

### PR DESCRIPTION
**Pull request by emilymye**
- Added new MM TF override flag `poll_for_existence`
- Added hook in resource template so TF polls for non-404 response post-create
- Supported in PubSub with test

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
pubsub: Added short poll for existance after creation to prevent a previously negative-cached result in read-after-create  
```

Part of https://github.com/terraform-providers/terraform-provider-google/issues/4993

### CODE REVIEW

**Comment by emilymye** [comment]
@ndmckinley we could add it to update but I thought it'd be a bit weird. Unless refresh is off, we do a read before update to make sure it still exists, so the existence check shouldn't start failing after update? I think if we run into a situation where we need it in update, we can add it later without much issue but not sure it makes sense right now

**Comment by ndmckinley** [comment]
Yeah, totally, I agree.  For eventually consistent updates, we'd want to ... hm, maybe check to see if a diff remained after the operation finished, basically?

We'd want to take the result from the operation and poll the Read endpoint until the result matched what the operation gave us, I _think._

Would that idea fit into this framework, or is there something else that might fit better?

**Comment by emilymye** [comment]
@ndmckinley that seems reasonable actually - maybe I'll go ahead and update this PR to match against the operation response instead.

**Comment by rileykarson** [comment]
@ndmckinley: Since that's already an MMv2 feature, I think we should defer that work until MMv2 unless we directly need it. If we don't have a pressing need to implement it yet, we can just defer those resources and add them as early MMv2 ones.

File: templates/terraform/resource.erb [hunk 1]
```diff
@@ -251,12 +251,53 @@ func resource<%= resource_name -%>Create(d *schema.ResourceData, meta interface{
 
     log.Printf("[DEBUG] Finished creating <%= object.name -%> %q: %#v", d.Id(), res)
 
+<% if object.poll_for_existence -%>
+    log.Printf("[DEBUG] Making sure eventually consistent <%= object.name -%> %q is ready", d.Id())
+    tryPollForResourceExistence(d, meta, resource<%= resource_name -%>ExistenceCheck)
```
[the remark this answers is not in the document]
Comment (emilymye) [reply templates/terraform/resource.erb]: yeah, i wanted the implication to be that this doesn't return the error, just tries and logs if it fails, so the behavior is fine if the actual Read completes? but i can change it to return an error

[the remark this answers is not in the document]
Comment (chrisst) [reply templates/terraform/resource.erb]: I'm in favor of return the error and allow the caller to suppress it if they don't care. I personally find methods that might do what I want and might not troubling.

**Comment by emilymye** [comment]
(gonna close this one and open a new one)

**Pull request closed without merging**